app/helpers/random_assign_helper.rb

```diff
@@ -0,0 +1,138 @@
+require 'set'
+
+class UnableToRandomlyAssignGroupException < Exception
+end
+
+module RandomAssignHelper
+  # Performs the random assignment, and stores them in the database.
+  # - pr_assignment: The peer review assignment.
+  # - num_groups_for_reviewers: How many PR's each group in reviewer_groups should have.
+  def perform_random_assignment(pr_assignment, num_groups_for_reviewers)
+    reviewer_groups_relation = pr_assignment.valid_groupings
+    reviewee_groups_relation = pr_assignment.parent_assignment.valid_groupings
+    @shuffled_reviewees = []
+    @num_groups_for_reviewers = num_groups_for_reviewers
+    @reviewer_ids_assigned_reviewee_ids_map = Hash.new { |h, k| h[k] = Set.new }
+    @eligible_reviewers = []
+    reviewer_groups_relation.each { |reviewer| @eligible_reviewers.push(reviewer) }
```

Comment:
Should I move the above into its own method to clean this part of the function up?